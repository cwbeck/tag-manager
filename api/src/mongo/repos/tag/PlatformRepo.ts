import Repo from '../../abstractions/Repo';
import { injectable } from 'inversify';
import Platform from '../../models/tag/Platform';

@injectable()
export default class PlatformRepo extends Repo<Platform> {
    protected readonly auditEnabled = true;
}
