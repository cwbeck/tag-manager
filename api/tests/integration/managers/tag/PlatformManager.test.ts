import ManagerTestUtils from '../../../utils/ManagerTestUtils';
import TestUtils from '../../../utils/TestUtils';
import PlatformRepo from '../../../../src/mongo/repos/tag/PlatformRepo';
import Mock = jest.Mock;

describe('PlatformManager', () => {
    beforeEach(ManagerTestUtils.beforeEachPrepareContainer);
    afterEach(TestUtils.afterEachRestoreContainerMocksAndDate);

    it('fetches a single platform', async () => {
        ManagerTestUtils.mockOrgUserWithViewAccess();
        ManagerTestUtils.mockOrgUserWithViewAccessOnPlatform();

        const mockSutModel = TestUtils.buildMockPlatform();
        const mockSutRepo = {} as PlatformRepo;
        mockSutRepo.findByIdThrows = jest.fn().mockResolvedValue(mockSutModel);
        (ManagerTestUtils.mockRepoFromModelFactory as Mock).mockReturnValue(mockSutRepo);

        const server = TestUtils.createApolloServerForIntegration();

        const Q = `
            query {
                getPlatform(id: "5edf4c48e9615d9a1c66bc2a") {
                    id
                    name
                }
            }
        `;

        // run query against the server and snapshot the output
        const res = await server.executeOperation({ query: Q });
        expect(res).toMatchSnapshot();
    });
});
