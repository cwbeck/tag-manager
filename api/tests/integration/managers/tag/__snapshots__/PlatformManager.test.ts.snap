// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlatformManager fetches a single platform 1`] = `
Object {
  "data": Object {
    "getPlatform": Object {
      "id": "5edf4c48e9615d9a1c66caaa",
      "name": "test platform",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
